lib/plugins/stats.js

```diff
@@ -0,0 +1,52 @@
+// Built-in plugin to expose stats about the deployment
+module.exports = async robot => {
+  const REFRESH_INTERVAL = 60 * 60 * 1000
+
+  // Cache of stats that get reported
+  const stats = {installations: 0, popular: []}
+
+  // Setup /stats endpoint to return cached stats
+  robot.router.get('/stats', (req, res) => {
+    res.json(stats)
+  })
+
+  // Refresh the stats when the plugin is loaded
+  await refresh()
+
+  // Refresh the stats on an interval
+  setInterval(refresh, REFRESH_INTERVAL)
+
+  async function refresh () {
+    const installations = await getInstallations()
+
+    stats.installations = installations.length
+    stats.popular = await popularInstallations(installations)
+  }
+
+  async function getInstallations () {
+    const github = await robot.auth()
+    const req = github.apps.getInstallations({per_page: 100})
+    return github.paginate(req, res => res.data)
+  }
+
+  async function popularInstallations (installations) {
+    let popular = await Promise.all(installations.map(async installation => {
+      const github = await robot.auth(installation.id)
+
+      const req = github.apps.getInstallationRepositories({per_page: 100})
+      const repositories = await github.paginate(req, res => {
+        return res.data.repositories.filter(repository => !repository.private)
+      })
+      const account = installation.account
+
+      account.stars = repositories.reduce((stars, repository) => {
+        return stars + repository.stargazers_count
```

Comment:
where is this `stars` coming from in both these lines? [Looking at the example payload](https://developer.github.com/v3/apps/installations/#list-repositories), it seems to me like the only data we'd need/use is `repository.stargazers_count`?

Reply:
It's the value being "reduced". It starts as `0`, which is passed in below, and then becomes the sum of `starts + stargazers_count`. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce?v=a